src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs

```diff
@@ -441,8 +456,19 @@ public async Task BeforeRemoval()
                 _ => throw new ArgumentOutOfRangeException()
             };
 
-            var translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y, 250);
-            await Task.Delay(250);
+            var length = 250;
+            Task translationTask = Task.CompletedTask;
+            if (HasAnticipationEffect)
+            {
+                length = 600;
+                translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y, (uint)length, Easing.SpringIn);
+            }
+            else
+            {
+                translationTask = m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y);

```

Comment:
This length is 250 right? So why is the Task.Delay 600 and not length?

Reply:
Length and task await should be 600 when anticipation is enabled

Reply:
I'll clean up when I will continue it the branch.